src/containers/NoteForm/NoteForm.js

```diff
@@ -72,16 +73,20 @@ export class NoteForm extends Component {
 
   handleSubmit = (e) => {
     e.preventDefault()
-    const { id, title } = this.state.note;
+    let { id, title } = this.state.note;
     let newItems = this.state.items;
     const { isEdit, editNote, postNote } = this.props;
     const url = isEdit ? `http://localhost:3001/api/v1/notes/${id}` : 'http://localhost:3001/api/v1/notes'
 
-    if (newItems[newItems.length - 1].description === '') {
+    if (newItems.length > 0 && newItems[newItems.length - 1].description === '') {
       newItems = newItems.filter(item => item.description !== '')
     }
-    isEdit ? editNote(url, { id, title, items: newItems }) : postNote(url, { id, title, items: newItems })
-    this.setState({ redirect: true, items: newItems })
+    if (title !== '') {
+      isEdit ? editNote(url, { id, title, items: newItems }) : postNote(url, { id, title, items: newItems })
+      this.setState({ redirect: true, items: newItems })
+    } else {
+      this.setState({ errorMessage: 'Please provide a title', items: newItems })
+    }
```

Comment:
Nice work giving explicit feedback to user as to what is missing from their form!